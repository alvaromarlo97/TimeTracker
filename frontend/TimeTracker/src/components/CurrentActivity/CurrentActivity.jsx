/* eslint-disable no-underscore-dangle */
/* eslint-disable global-require */
/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-no-bind */
import React from 'react';
import {
  SafeAreaView,
  Text,
  TouchableOpacity,
} from 'react-native';
import { useSelector } from 'react-redux';
import styles from './currentActivities.style';

export default function CurrentActivity() {
  const activity = useSelector(({ loadActivity }) => loadActivity?.activityTime);

  return (
    <SafeAreaView style={styles.container}>

      {activity?.map((element) => (
        <TouchableOpacity style={styles.button} key={element._id}>
          <Text>{element.hours}</Text>
        </TouchableOpacity>
      ))}

    </SafeAreaView>
  );
}
