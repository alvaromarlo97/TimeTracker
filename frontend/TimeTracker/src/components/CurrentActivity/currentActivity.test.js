import React from 'react';
import { render } from '../../../utils/testUtils/test.utils';
import CurrentActivity from './CurrentActivity';

const navigation = {
  navigate: jest.fn(),
};

describe('Given a Character Component', () => {
  describe('When it is rendered', () => {
    let screen;
    beforeEach(() => {
      screen = render(<CurrentActivity navigation={navigation} />);
    });
    test('Then screen should match the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });
  });
});
