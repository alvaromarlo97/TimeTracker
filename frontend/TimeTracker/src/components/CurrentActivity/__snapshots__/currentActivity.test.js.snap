// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Character Component When it is rendered Then screen should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(235,255,254,1)",
      "display": "flex",
      "flex": 1,
      "height": 700,
    }
  }
/>
`;
